Replicator/c4Socket.cc

```diff
@@ -145,16 +156,22 @@ namespace litecore { namespace repl {
         WebSocketImpl::connect();
         if (_factory.open) {
             net::Address c4addr(url());
-            _factory.open(this, &c4addr, options().data(), _factory.context);
+            try {
```

Comment:
I think the try/catch belongs in the factory implementations. They should not throw exceptions.

Reply:
In general I am against putting trust in code that we don't control if it has the ability to cascade through our base like this....but I am willing to accept it under the argument of "This is not meant for public consumption and you do so at your own risk"